File: EPSILON/TOOLS/Altova/Altova/XMLSpy2006/spl/java/altova/text/tablelike/Table.java
[call GenerateFileHeader("Table.java")]
package com.altova.text.tablelike;

import java.util.ArrayList;

import com.altova.AltovaException;

public abstract class Table {
    private ArrayList m_Records = new ArrayList();

    private Header m_Header = new Header();

    private ISerializer m_Serializer = null;

    protected abstract ISerializer createSerializer();

    protected abstract void initHeader(Header header);

    public Table() {
        this.initHeader(m_Header);
        m_Serializer = this.createSerializer();
    }

    public Header getHeader() {
        return m_Header;
    }

    public int size() {
        return m_Records.size();
    }

    public void add(Record rhs) {
        m_Records.add(rhs);
    }

    public void clear() {
        m_Records.clear();
    }

    public Record getAt(int index) {
        return (Record) m_Records.get(index);
    }

    public String getEncoding() {
        return m_Serializer.getCharset();
    }

    public void setEncoding(String rhs) {
        m_Serializer.setCharset(rhs);
    }

    public void save(String filename) {
        try {
            m_Serializer.serialize(filename);
        } catch (MappingException x) {
            throw new AltovaException(x);
        }
    }

    public void parse(String filename) {
        try {
            m_Serializer.deserialize(filename);
        } catch (MappingException x) {
            throw new AltovaException(x);
        }
    }
}

File: EPSILON/SAMPLE FILE/E-publishing/Tools/Altova/Altova/XMLSpy2006/spl/cpp/AltovaText/Table.cpp
[call GenerateFileHeader("Table.cpp")]
#include "stdafx.h"
#include "Table.h"
#include "Helpers.h"
#include "Serializer.h"
#include "Record.h"
#include "Header.h"
#include "RecordBasedParser.h"
#include "../Altova/AltovaException.h"

namespace altova
{
namespace text
{
namespace tablelike
{

CTable::CTable()
:	m_pSerializer(NULL)
{
}

CTable::~CTable()
{
	delete m_pSerializer;
	m_pSerializer = NULL;

	Clear();
}

void CTable::Parse(const tstring& filename)
{
	try
	{
		Setup();

		m_pSerializer->Deserialize(filename);
	}
	catch (CMappingException& x)
	{
		throw altova::CAltovaException(0, filename + _T(":") + x.GetMessage());
	}
}

void CTable::Save(const tstring& filename)
{
	tstringstream stream;
	
	Setup();
	
	m_pSerializer->SetStream(stream);
	m_pSerializer->Serialize();
	
	SaveFileFromBuffer(filename, stream.str(), m_nEncoding, m_nByteorder, m_bBom);
}

size_t CTable::GetRecordCount() const
{
	return m_Records.size();
}

const CRecord& CTable::GetRecordAt(size_t index) const
{
	return *m_Records\[index\];
}

void CTable::AddRecord(const CRecord& rhs)
{
	m_Records.push_back(new CRecord(rhs));
}

void CTable::Clear()
{
	DeleteContainedPointers( m_Records );
	m_Records.clear();
}

CHeader& CTable::GetHeader()
{
	return m_Header;
}

void CTable::Setup()
{
	if( NULL == m_pSerializer )
	{
		InitHeader(m_Header);
		
		m_pSerializer = CreateSerializer();
	}
}

} // namespace tablelike
} // namespace text
} // namespace altova
